src/main/java/uk/gov/hmcts/reform/idam/service/IdamUserRestorerService.java

```diff
@@ -33,11 +33,16 @@ public void run() {
 
         securityUtil.generateTokens();
         int requestsMade = 0;
-
+        int startPage = restoreSummary.getStartPage();
         while (lauService.hasMoreRecords() && requestsMade < requestsLimit) {
             List<DeletionLog> deletedUsers = lauService.fetchDeletedUsers();
-            log.info("Fetched deleted users {}", deletedUsers.stream().map(DeletionLog::getUserId).toList());
-            requestsMade++;
+            int currentPage = startPage + requestsMade;
+            log.info(
+                "[Page {}] Fetched deleted users {}",
+                currentPage,
+                deletedUsers.stream().map(DeletionLog::getUserId).toList()
+            );
+            restoreSummary.addRequestsMade(++requestsMade);
```

Comment:
I don't like inlining ++ operation as you need to understand when it does the operation. Since we start at 0, adding ++ after would log as 0 and only then increased it. Adding in front, it increases first and then logs it. Now that I think about it, this can also be set outside of the loop - similar how we call setEndTime().

Reply:
Looking at the if the requestMade is 0 , it will log 1 , is this we want to achieve or we want current requestMade . We need to log first and then increment